src/lib.rs

```diff
@@ -249,9 +249,15 @@ pub struct RegisterArrayInfo {
 }
 
 #[derive(Clone, Debug)]
+pub struct RegisterClusterInfo {
+    pub registers: Option<Vec<Register>>,
+}
+
+#[derive(Clone, Debug)]
 pub enum Register {
     Single(RegisterInfo),
     Array(RegisterInfo, RegisterArrayInfo),
+    Cluster(RegisterInfo, RegisterClusterInfo),
```

Comment:
I think I would prefer a separate `Cluster` type instead of adding a variant here. Possibly with these fields: `struct Cluster { info: RegisterInfo, chidren: Either<Register, Cluster> }`.

`Register` should be used only for `<register>` trees.